bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/NashornScriptEngineFactory.java

```diff
@@ -12,74 +12,67 @@
 
-import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Set;
 
 import javax.script.ScriptEngine;
-import javax.script.ScriptEngineManager;
 import javax.script.ScriptException;
 
 import org.openhab.core.automation.module.script.ScriptEngineFactory;
 import org.osgi.service.component.annotations.Component;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
+ * An implementation of {@link ScriptEngineFactory} with customizations for Nashorn ScriptEngines.
  *
  * @author Simon Merschjohann - Initial contribution
+ * @author Scott Rushworth - removed default methods provided by ScriptEngineFactory
  */
 @Component(service = ScriptEngineFactory.class)
-public class NashornScriptEngineFactory implements ScriptEngineFactory {
-    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+public class NashornScriptEngineFactory extends AbstractScriptEngineFactory {
 
-    private ScriptEngineManager engineManager = new ScriptEngineManager();
+    private static final String SCRIPT_TYPE = "js";
 
     @Override
-    public List<String> getLanguages() {
-        return Arrays.asList("js", "javascript", "application/javascript");
+    public void logFactoryDetails() {
+        logger.info("Added customized ScriptEngineFactory for Nashorn");
     }
 
     @Override
-    public void scopeValues(ScriptEngine engine, Map<String, Object> scopeValues) {
-        Set<String> expressions = new HashSet<String>();
+    public List<String> getScriptTypes() {
+        List<String> scriptTypes = new ArrayList<>();
 
-        for (Entry<String, Object> entry : scopeValues.entrySet()) {
-            engine.put(entry.getKey(), entry.getValue());
+        for (javax.script.ScriptEngineFactory f : engineManager.getEngineFactories()) {
+            List<String> extensions = f.getExtensions();
 
-            if (entry.getValue() instanceof Class) {
-                expressions.add(String.format("%s = %s.static;", entry.getKey(), entry.getKey()));
+            if (extensions.contains(SCRIPT_TYPE)) {
+                scriptTypes.addAll(extensions);
+                scriptTypes.addAll(f.getMimeTypes());
             }
         }
-        String scriptToEval = String.join("\n", expressions);
-        try {
-            engine.eval(scriptToEval);
-        } catch (ScriptException e) {
-            logger.error("ScriptException while importing scope: {}", e.getMessage());
-        }
+        logger.trace("getScriptTypes(): {}", scriptTypes);
+        return scriptTypes;
     }
 
     @Override
-    public ScriptEngine createScriptEngine(String fileExtension) {
-        ScriptEngine engine = engineManager.getEngineByExtension(fileExtension);
+    public void scopeValues(ScriptEngine scriptEngine, Map<String, Object> scopeValues) {
+        Set<String> expressions = new HashSet<String>();
```

Comment:
You can change the second `<String>` to `<>`.

I would expect the save action called formatter replaces such stuff.

Reply:
Done. This was the original code... it only showed in the diff because I moved the method.